pkg/sriov/sriov.go

```diff
@@ -152,31 +170,55 @@ func (s *sriovManager) SetupVF(conf *sriovtypes.NetConf, podifName string, netns
 			}
 		}
 
-		logging.Debug("8. Enable Optimistic DAD for IPv6 addresses", "func", "SetupVF",
-			"linkObj", linkObj)
+		logging.Debug("7. Enable Optimistic DAD for IPv6 addresses", "func", "SetupVF",
+			"linkObj", netNSLinkObj)
 		_ = s.utils.EnableOptimisticDad(podifName)
 
-		// 9. Bring IF up in Pod netns
-		logging.Debug("9. Bring IF up in Pod netns",
+		// 8. Bring IF up in Pod netns
+		logging.Debug("8. Bring IF up in Pod netns",
 			"func", "SetupVF",
-			"linkObj", linkObj)
-		if err := s.nLink.LinkSetUp(linkObj); err != nil {
+			"linkObj", netNSLinkObj)
+		if err = s.nLink.LinkSetUp(netNSLinkObj); err != nil {
 			return fmt.Errorf("error bringing interface up in container ns: %q", err)
 		}
 
 		return nil
-	}); err != nil {
+	})
+	if err != nil {
 		return fmt.Errorf("error setting up interface in container namespace: %q", err)
 	}
 
 	// Copy the MTU value to a new variable
 	// and use it as a pointer
 	vfMTU := linkObj.Attrs().MTU
 	conf.MTU = &vfMTU
-
 	return nil
 }
 
+// moveAndRenameLink restores the original link name and moves it back to initns
+func (s *sriovManager) moveAndRenameLink(sourceNM ns.NetNS, podifName, linkName string, initns ns.NetNS) error {
+	err := sourceNM.Do(func(_ ns.NetNS) error {
+		// Restore the original link name in case of error in renaming
+		if nsLinkObj, err := s.nLink.LinkByName(podifName); err == nil {
```

Comment:
if we failed to rename the link, then it would be with its original name (linkName) in this case we should also try and move it to the initns